overlord/snapstate/backend/setup.go

```diff
@@ -82,11 +82,8 @@ func (b Backend) SetupSnap(snapFilePath, instanceName string, sideInfo *snap.Sid
 		return snapType, nil, err
 	}
 
-	if s.InstanceKey != "" {
-		err := os.MkdirAll(snap.BaseDir(s.SnapName()), 0755)
-		if err != nil && !os.IsExist(err) {
-			return snapType, nil, err
-		}
+	if err := createSharedSnapDirForParallelInstance(s); err != nil {
+		return snapType, nil, err
 	}
```

Comment:
shouldn't we leave a comment on why we do this in two places now?

Reply:
left a comment